File: weekly_397th/A.go
func findPermutationDifference(s string, t string) int {
    ans := 0
    arr := [26]int{}
    for i, c := range s {
        arr[c - 'a'] = i
    }

    for i, c := range t {
        ans += int(math.Abs(float64(arr[c - 'a'] - i)))
    }
    return ans
}

File: weekly_397th/A.cpp
class Solution {
public:
    int findPermutationDifference(string s, string t) {
        vector<int> cnt(26);
        for(int i = 0; i < s.size(); i++) {
            cnt[s[i] - 'a'] = i;
        }

        int res = 0;
        for(int i = 0; i < t.size(); i++) {
            res += abs(cnt[t[i] - 'a'] - i);
        }
        return res;
    }
};
